### ISSUE TITLE
Regression: false positive `inject-field`

### ISSUE THREAD

**Issue by clason** [EmmyLuaLs/emmylua-analyzer-rust#660]
caf99943558f9bc388e5b6a16f6b6bdd1bb94f18 introduced a regression for Neovim:
```
warning: Fields cannot be injected into the reference of `vim.var_accessor` for `loaded_nvim_treesitter`.  [inject-field]
  -->: nvim-treesitter.lua:4:7

  3 │ end
  4 │ vim.g.loaded_nvim_treesitter = true
  5 │
```
(this is from nvim-treesitter, main branch, `plugin/nvim-treesitter.lua`, but any similar `vim.g.foo = ...` will trigger this).

For reference, `vim.g` is defined here: https://github.com/neovim/neovim/blob/b0e8b0a35f3d38763c564f6d4924f7be2e2271fb/runtime/lua/vim/_editor.lua#L413-L435
